modules/cloud-run/variables.tf

```diff
@@ -0,0 +1,152 @@
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "audit_log_triggers" {
+  description = "Event arc triggers (Audit log)"
+  type = list(object({
+    service_name = string
+    method_name  = string
+  }))
+  default = null
+}
+
+variable "containers" {
+  description = "Containers"
+  type = list(object({
+    image   = string
+    command = list(string)
+    args    = list(string)
+    env     = map(string)
+    env_from = map(object({
+      key  = string
+      name = string
+    }))
+    resources = object({
+      limits = object({
+        cpu    = string
+        memory = string
+      })
+      requests = object({
+        cpu    = string
+        memory = string
+      })
+    })
+    ports = list(object({
```

Comment:
Would it make sense to have this as name => {protocol, container_port}?

Reply:
According to the APi all three are optional that is why I put 3
https://cloud.google.com/run/docs/reference/rest/Shared.Types/Container#ContainerPort

Reply:
I would skip protocol, as seems to just be a placeholder for future features: `Protocol for port. Must be "TCP". Defaults to "TCP"`.

Reply:
done